internal/cloud/backend_runTasks.go

```diff
@@ -138,12 +181,150 @@ func (b *Cloud) runTasksWithTaskResults(context *IntegrationContext, output Inte
+
+		if err != nil {
+			return false, generalError("Failed to retrieve task stage", err)
+		}
+
+		if len(stage.PolicyEvaluations) == 0 {
+			return false, nil
+		}
+
+		summary := summarizePolicyEvaluationResults(stage.PolicyEvaluations)
+		if summary.unreachable {
+			output.Output("Skipping policy evaluation.")
+			output.End()
+			return false, nil
+		}
+
+		if summary.pending > 0 {
+			pendingMessage := "Evaluating ... "
+
+			if i%4 == 0 {
+				if i > 0 {
+					output.OutputElapsed(pendingMessage, len(pendingMessage))
+				}
+			}
+			return true, nil
+		}
+
+		// No more policy evaluations pending/running. Print all the results.
+		output.Output("\n------------------------------------------------------------------------\n")
+		output.Output("[bold]OPA Policy Evaluation\n")
+
+		var result, message string
+		// Currently only one policy evaluation supported : OPA
+		if stage.PolicyEvaluations[0].Status == tfe.PolicyEvaluationPassed {
+			message = "[dim] This result means that all OPA policies passed and the protected behaviour is allowed"
+			result = fmt.Sprintf("[green]%s", strings.ToUpper(string(tfe.PolicyEvaluationPassed)))
+			if stage.PolicyEvaluations[0].ResultCount.AdvisoryFailed > 0 {
+				result += " (with advisory)"
+			}
+		} else {
+			message = "[dim] This result means that one or more OPA policies failed. More than likely, this was due to the discovery of violations by the main rule and other sub rules"
+			result = fmt.Sprintf("[red]%s", strings.ToUpper(string(tfe.PolicyEvaluationFailed)))
+		}
+
+		output.Output(fmt.Sprintf("[bold]%c%c Overall Result: %s", Arrow, Arrow, result))
+
+		output.Output(message)
+
+		total := getPolicyCount(stage.PolicyEvaluations[0].ResultCount)
+
+		output.Output(fmt.Sprintf("%d policies evaluated\n", total))
+
+		for _, policyEvaluation := range stage.PolicyEvaluations {
+			policyOutcomes, err := b.client.PolicySetOutcomes.List(context.StopContext, policyEvaluation.ID, nil)
+			if err != nil {
+				return false, err
+			}
+
+			for i, out := range policyOutcomes.Items {
+				output.Output(fmt.Sprintf("%c Policy set %d: [bold]%s (%d)", Arrow, i+1, out.PolicySetName, len(out.Outcomes)))
+				for _, outcome := range out.Outcomes {
+					output.Output(fmt.Sprintf("  %c Policy name: [bold]%s", DownwardArrow, outcome.PolicyName))
+					switch outcome.Status {
+					case "passed":
+						output.Output(fmt.Sprintf("     | [green][bold]%c Passed", Tick))
+					case "failed":
+						if outcome.EnforcementLevel == "advisory" {
+							output.Output(fmt.Sprintf("     | [blue][bold]%c Advisory", Warning))
+						} else {
+							output.Output(fmt.Sprintf("     | [red][bold]%c Failed", Cross))
+						}
+					}
+					if outcome.Description != "" {
+						output.Output(fmt.Sprintf("     | [dim]%s", outcome.Description))
+					} else {
+						output.Output("     | [dim]No description available")
+					}
+				}
+			}
+		}
+
+		stage, err = fetchTaskStage(b, context.StopContext)
```

Comment:
Curious why we're fetching the task stage again. It feels like the call on line 214 should still be valid

Reply:
So I noticed while testing that the task stage sometimes has the status "running" (instead of awaiting_override) even when the policy has finished evaluation. This is very rare though, however, I am fetching the status here again to avoid using a stale status.

Reply:
Wouldn't the next poll pick that up though? Actually this may be an opportunity to restructure this.

In theory an "Awaiting Override" event happens independent of task result or task stage. Probably best to chat over zoom for this.